edge/pkg/devicetwin/dtmanager/dmiserver.go

```diff
@@ -0,0 +1,412 @@
+/*
+Copyright 2022 The KubeEdge Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package dtmanager
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"strings"
+
+	"k8s.io/klog/v2"
+
+	"github.com/kubeedge/beehive/pkg/core/model"
+	"github.com/kubeedge/kubeedge/cloud/pkg/devicecontroller/constants"
+	pb "github.com/kubeedge/kubeedge/edge/pkg/apis/dmi/v1"
+	"github.com/kubeedge/kubeedge/edge/pkg/devicetwin/dmiclient"
+	"github.com/kubeedge/kubeedge/edge/pkg/devicetwin/dmiserver"
+	"github.com/kubeedge/kubeedge/edge/pkg/devicetwin/dtcommon"
+	"github.com/kubeedge/kubeedge/edge/pkg/devicetwin/dtcontext"
+	"github.com/kubeedge/kubeedge/edge/pkg/devicetwin/dttype"
+	"github.com/kubeedge/kubeedge/edge/pkg/metamanager/dao"
+	"github.com/kubeedge/kubeedge/pkg/apis/devices/v1alpha2"
+)
+
+//TwinWorker deal twin event
+type DMIWorker struct {
+	Worker
+	Group string
+}
+
+var (
+	//dmiActionCallBack map for action to callback
+	dmiActionCallBack map[string]CallBack
+
+	mapperList      map[string]*pb.MapperInfo
+	deviceModelList map[string]*v1alpha2.DeviceModel
+	deviceList      map[string]*v1alpha2.Device
```

Comment:
the three map need add a mutex lock

Reply:
done. i have move these 3 map to the `DMIWorker` struct and make 3 mutex for them.